core-bundle/src/Resources/contao/modules/ModuleSearch.php

```diff
@@ -97,7 +97,7 @@ protected function compile()
 		$this->Template->results = '';
 
 		// Execute the search if there are keywords
-		if ((string) $strKeywords !== '' && $strKeywords != '*' && !$this->jumpTo)
+		if ($strKeywords !== '' && $strKeywords != '*' && !$this->jumpTo)
```

Comment:
If `$strKeywords` is always a string (and it should be because `trim` always outputs a string) - shouldn't the second comparison also be strict?

Reply:
No. The first comparison is meant to be type safe, the second one is not. Of course, we know that `$strKeywords` is a string, so `!==` would work as well. But since it does not matter, there is no point in changing it. 🤷🏻‍♂️